src/antsibull_changelog/copier.py

```diff
@@ -0,0 +1,129 @@
+# Author: Felix Fontein <felix@fontein.de>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or
+# https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: 2020, Ansible Project
+
+"""
+Directory and collection copying helpers.
+"""
+
+from __future__ import annotations
+
+import os
+import shutil
+import subprocess
+import tempfile
+
+from .config import ChangelogConfig, PathsConfig
+from .logger import LOGGER
+from .utils import detect_vcs
+
+
+class Copier:  # pylint: disable=too-few-public-methods
+    """
+    Allows to copy directories.
+    """
+
+    def copy(self, from_path: str, to_path: str) -> None:
```

Comment:
Can this be modified to accept any `StrPath`?

Reply:
That would work, but I don't really want to add typeshed to the project just for `StrPath`.

Reply:
Well, it's put under a TYPE_CHECKING block. You can also do `str | os.PathLike[str]`.